### PR TITLE
Disable time tracking in wasm release builds because it's expensive

### PR DESCRIPTION
**Pull request by kg**
Right now on main we spend a good chunk of cpu time just calling _emscripten_get_now in order to measure how long interpreter code generation takes. On most targets the equivalent timestamp checks are fast but on wasm, they're not.

So this PR disables those checks in wasm release builds specifically. Based on my measurements this should improve startup time a little bit.

### CODE REVIEW

**Review by vargaz — APPROVED** [review]

**Review by steveisok — APPROVED** [review]

File: src/mono/mono/mini/mini.h [hunk 1]
```diff
@@ -2811,12 +2811,20 @@ void mono_cfg_add_try_hole (MonoCompile *cfg, MonoExceptionClause *clause, guint
 void mono_cfg_set_exception (MonoCompile *cfg, MonoExceptionType type);
 void mono_cfg_set_exception_invalid_program (MonoCompile *cfg, const char *msg);
 
+#if defined(HOST_WASM) && !(defined(DEBUG) || defined(_DEBUG))
```
Comment (lambdageek) [inline src/mono/mono/mini/mini.h:2814]: do we really set these in debug builds for mono?

I can see in mono.proj that we set them on windows sometimes, and also in android builds.  but not elsewhere.

Comment (kg) [reply src/mono/mono/mini/mini.h:2814]: That's a great question, I don't know how to answer it

Comment (lambdageek) [reply src/mono/mono/mini/mini.h:2814]: @kg 

```c
# if defined(HOST_WASM) && !(defined (DEBUG) || defined(_DEBUG)
...
#elif defined(HOST_WASM)
# error "this shouldn't happen on Release lanes"
#else
...
#endif
```

and see which lanes fail to build?

Comment (vargaz) [reply src/mono/mono/mini/mini.h:2814]: Might be better to just use #ifdef HOST_WASM, and enable it manually if somebody needs it.

**Review by lambdageek — APPROVED** [review]

**Comment by kg** [comment]
This passed CI a couple times previously, so even though the lanes aren't working right now, is it fine to merge it?

**Review by maraf — APPROVED** [review]

**Comment by radekdoulik** [comment]
nice, it helped with the startup indeed https://radekdoulik.github.io/WasmPerformanceMeasurements/?startDate=2024-03-12T10%3A19%3A36.040Z&endDate=2024-03-19T10%3A19%3A36.040Z&tasks=AppStart

image

**Pull request merged**